src/v/raft/recovery_stm.cc

```diff
@@ -466,10 +486,10 @@ ss::future<> recovery_stm::apply() {
     return ss::with_gate(
              _ptr->_bg,
              [this] {
-                 return do_recover().then([this] {
+                 return recover().then([this] {
                      return ss::do_until(
                        [this] { return is_recovery_finished(); },
-                       [this] { return do_recover(); });
+                       [this] { return recover(); });
```

Comment:
Is there a reason this isn't:
```cpp
                 return ss::do_until(
                   [this] { return is_recovery_finished(); },
                   [this] { return recover(); });
```

Reply:
we need at least one pass of recovery to decide if it is finished. it is `do {} while(...)` now, instead of `while(...){}`